p2p/peer_set.go

```diff
@@ -119,25 +119,27 @@ func (ps *PeerSet) Remove(peer Peer) bool {
 		return false
 	}
 
+	if len(ps.list) == 0 {
+		return false
```

Comment:
Shouldn't we panic here? I mean, if the `item` is not `nil`, then there must be at least 1 peer in `ps.list`, no?

Reply:
```suggestion
                peer.SetRemovalFailed()
		return false
```

at the very least we should call `peer.SetRemovalFailed()` (for same reasons as above)

Reply:
Gotch and thank you @melekes! Updated